src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -379,7 +381,14 @@ void RealmCoordinator::do_get_realm(RealmConfig&& config, std::shared_ptr<Realm>
     // rerun_on_open was set
     if (subscription_function) {
         const auto current_subscription = realm->get_latest_subscription_set();
-        if ((current_subscription.version() == 0) || (rerun_on_open && db_opened_first_time)) {
+        const auto subscription_version = current_subscription.version();
+        // in case we are hitting this check while during a normal open, we need to take in
+        // consideration if the db was created during this call. Since this may be the first time
+        // we are actually creating a realm. For async open this does not apply, infact db_created
+        // will always be false.
+        first_time_open |= db_created;
```

Comment:
`|=` on bools is weird. It works, but using a bitwise operation is really misleading as it suggests that bitflags on an int are being used rather than bools.

Reply:
OK, let's go with a more explicit check then.
```
if(!first_time_created)
   first_time_created = db_created;
```